db-scripts/src/main/resources/cgds.sql

```diff
@@ -285,14 +285,16 @@ CREATE TABLE `genetic_profile` (
   `GENETIC_PROFILE_ID` int(11) NOT NULL auto_increment,
   `STABLE_ID` varchar(255) NOT NULL,
   `CANCER_STUDY_ID` int(11) NOT NULL,
+  `REFERENCE_GENOME_ID` int(4) NOT NULL,
   `GENETIC_ALTERATION_TYPE` varchar(255) NOT NULL,
   `DATATYPE` varchar(255) NOT NULL,
   `NAME` varchar(255) NOT NULL,
   `DESCRIPTION` mediumtext,
   `SHOW_PROFILE_IN_ANALYSIS_TAB` BOOLEAN NOT NULL,
   PRIMARY KEY (`GENETIC_PROFILE_ID`),
-  UNIQUE (`STABLE_ID`),
-  FOREIGN KEY (`CANCER_STUDY_ID`) REFERENCES `cancer_study` (`CANCER_STUDY_ID`) ON DELETE CASCADE
+  UNIQUE (`STABLE_ID`, `REFERENCE_GENOME_ID`),
```

Comment:
So genetic profiles won't be unique by stableId anymore? This is a big change, might cause a lot of complications. @jjgao

Reply:
Agreed with @ersinciftci. STABLE_ID should be unique.

Reply:
@khzhu do you have any use case that the same stable ID is required for different genome builds in the same study?

Reply:
@jjgao @ersinciftci , how about we set a rule: per study per genome build? The plan was to support multiple genome profiling within the same study.

Reply:
I don't know much about this new feature, but it feels like we should do this by using a relationship table, instead of changing the genetic_profile table. So one profile would have relations with more than one genome profile (or build? not sure what it means), but it still would be unique by stable id.

Reply:
We can do more normalization by introducing a new cross-reference genome profile table. But that would require quite changes in the code base, since the frontend would have to allow users to select the genome build of interest (having a reference genome dropdown box for instance), and then web services would also need to be changed to allow query the genetic profile based on the genome selection.

Reply:
As I said I don't know about the functionality much, but we have to change the whole application if genetic_profile can't be system-wise unique by the STABLE_ID anymore.

Reply:
@ersinciftci @jjgao ,please find revised schema diagram. I've added a new table named **reference_genome_profile** as a cross reference table for **genetic_profile** and **reference_genome** table. Now, **genetic_profile** can still be uniquely identified by **STABLE_ID**. In case of multiple genome profiling, then **reference_genome_profile** table will be used as a look up. 
![reference_genome_profile](https://user-images.githubusercontent.com/17375786/34847493-7da71dda-f6e9-11e7-8234-e8184fccb9c7.png)

Reply:
@khzhu   I don't think we need reference_genome_profile table because one profile will only have one reference_genome_id. I think we should just change the current to "UNIQUE (`STABLE_ID`)".

Reply:
@khzhu what do you mean by multiple genome profiling? Do you mean, for example, mutations called by both grch38 and grch37?

Reply:
yes, @jjgao, do we leave **REFERENCE_GENOME_ID** in the genetic_profile table as it is now just remove it from the unique constraint?

Reply:
@khzhu in that case, should we have different profiles for different reference genomes, e.g. study_id_mutations_grch37 and study_id_mutations_grch38? Why do we need to have them in the same profile?

Reply:
@jjgao , the reference genome used in profiling is a very important piece of information once the system had multiple reference genome datasets, having hg19 datasets coexists with hg38 datasets for example. I would suggest to have it properly tracked in a table. 
Since we already had a rule, per profile per genome, that ruled out having a new reference_genome_profile table (Normally, one to one relation does not be tracked in a different table). Besides, both copy number alteration and segment tables (Third PR) will have to use the reference genome to pull out correct cytobands or genomic coordinate for display. Currently, the reference genome information used by the copy number segments is tracked in copy_number_seg_file table, that table is going to be deprecated soon. 
Also, the frontend QueryStore.ts file expected the format of STABLE_ID is in a format of **study_id+genetic_alteration_type**, adding a reference genome suffix might have more impact than adding a new **reference_genome_id** field since it only affect importer scripts.

Reply:
@khzhu do you know what line in QueryStore.ts expects a geneticProfileId to be in `study_id+genetic_alteration_type` format? Because it must be a mistake, we should never assume that.

Reply:
@ersinciftci , line 946 ~ 979.

Reply:
@ersinciftci, not geneticProfileId but sampleListID.

Reply:
@khzhu @ersinciftci: I think I am confused -- may we all are :). Let's see if we can agree on the followings:

1. The profile stable ID should be unique and therefore it should be `UNIQUE(STABLE_ID)`
2. One profile should only have one reference genome version, and therefore it is 1-to-N between genetic_profile and reference_genome table. In that case, we don't need reference_genome_profile table, and reference_genome_id will reside in genetic_profile table.
3. The frontend should not make any assumption about profile stable ID. We definitely have multiple profiles per alteration_type in a study.

Are you agree with the above?

Reply:
@jjgao , crystal clear, I could not agree more :).

Reply:
@jjgao `One profile should only have one reference genome version, and therefore it is 1-to-N between genetic_profile and reference_genome table.` I'm confused now :) you mean 1-to-1 (instead of 1-to-N)?

Reply:
@ersinciftci I think it's 1-N (or 1-M) because 1 profile can only have 1 genome but 1 genome may exist in multiple profiles. It is similar to Employee and Department relationship in this post (http://www.oracle.com/technetwork/issue-archive/2011/11-nov/o61sql-512018.html).

Reply:
@jjgao Ah I see, ok it's clear to me too now.